packages/melody-streams/src/operators/createState.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright 2019 trivago N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BehaviorSubject } from 'rxjs';
+
+export const createState = initialValue => {
+    const subj = new BehaviorSubject(initialValue);
+
+    const mutator = newValue => {
```

Comment:
I like the extra affordance of this particular overload and believe it could enable some very nice patterns (though I'm not sure yet which those would be). Adding it later would not be an option. Removing it also won't be possible.

Would you object strongly against keeping it? Or is it rather a preference?

Reply:
If you ask me, this is a quite useful feature, e.g. for implementing a state reducer.

Reply:
I also believe this is a super useful thing. Having an option to pass in a function makes the API quite flexible and super powerful. I would strongly advise to keep it.